src/sentry/rules/conditions/event_frequency.py

```diff
@@ -278,6 +284,39 @@ def query_hook(
         )
         return sums[event.group_id]
 
+    def get_sums(
+        self, model: int, groups: list[int], start: datetime, end: datetime, environment_id: int
+    ) -> dict[int, int]:
+        sums: Mapping[int, int] = self.tsdb.get_sums(
+            model=get_issue_tsdb_group_model(model),
+            keys=[group.id for group in groups],
+            start=start,
+            end=end,
+            environment_id=environment_id,
+            use_cache=True,
+            jitter_value=groups[0].id,
+            tenant_ids={"organization_id": groups[0].project.organization_id},
+            referrer_suffix="alert_event_frequency",
+        )
+        return sums
+
+    def batch_query_hook(
+        self, group_ids: list[int], start: datetime, end: datetime, environment_id: str
+    ) -> dict[int, int]:
+        batch_sums: dict[int, int] = defaultdict(int)
+        groups = Group.objects.filter(id__in=group_ids)
+        error_issues = [group for group in groups if group.issue_category == GroupCategory.ERROR]
+        generic_issues = [group for group in groups if group.issue_category != GroupCategory.ERROR]
+
+        for group_chunk in chunked(error_issues, SNUBA_LIMIT):
+            error_sums = self.get_sums(GroupCategory.ERROR, group_chunk, start, end, environment_id)
+            batch_sums.update(error_sums)
+
+        for group_chunk in chunked(generic_issues, SNUBA_LIMIT):
+            generic_sums = self.get_sums("", group_chunk, start, end, environment_id)
+            batch_sums.update(generic_sums)
```

Comment:
Could we make this a bit more DRY?

We should also add some checks to make sure we don't invoke get_sums if the group_chunk is empty - 🤞 that tsdb does that under the hood, but who knows.